x-pack/plugins/security_solution/public/management/components/artifact_list_page/artifact_list_page.tsx

```diff
@@ -0,0 +1,292 @@
+    } = useWithArtifactListData(apiClient, searchableFields);
+
+    const items = useMemo(() => {
+      return listDataResponse?.data ?? [];
+    }, [listDataResponse?.data]);
+
+    const [selectedItemForDelete, setSelectedItemForDelete] = useState<
+      undefined | ExceptionListItemSchema
+    >(undefined);
+
+    const [selectedItemForEdit, setSelectedItemForEdit] = useState<
+      undefined | ExceptionListItemSchema
+    >(undefined);
+
+    const labels = useMemo<typeof artifactListPageLabels>(() => {
+      return {
+        ...artifactListPageLabels,
+        ..._labels,
+      };
+    }, [_labels]);
+
+    const handleOnCardActionClick = useCallback<UseArtifactCardPropsProviderProps['onAction']>(
+      ({ type, item }) => {
+        switch (type) {
+          case 'edit':
+            setSelectedItemForEdit(item);
+            setUrlParams({ show: 'edit', itemId: item.item_id });
+            break;
+
+          case 'delete':
+            setSelectedItemForDelete(item);
+            break;
+        }
+      },
+      [setUrlParams]
+    );
+
+    const handleCardProps = useArtifactCardPropsProvider({
+      items,
+      onAction: handleOnCardActionClick,
+      cardActionDeleteLabel: labels.cardActionDeleteLabel,
+      cardActionEditLabel: labels.cardActionEditLabel,
+      dataTestSubj: getTestId('card'),
+    });
+
+    const policiesRequest = useGetEndpointSpecificPolicies({
+      onError: (err) => {
+        toasts.addWarning(getLoadPoliciesError(err));
+      },
+    });
+
+    const memoizedRouteState = useMemoizedRouteState(routeState);
+
+    const backButtonEmptyComponent = useMemo(() => {
+      if (memoizedRouteState && memoizedRouteState.onBackButtonNavigateTo) {
+        return <BackToExternalAppSecondaryButton {...memoizedRouteState} />;
+      }
+    }, [memoizedRouteState]);
+
+    const backButtonHeaderComponent = useMemo(() => {
+      if (memoizedRouteState && memoizedRouteState.onBackButtonNavigateTo) {
+        return <BackToExternalAppButton {...memoizedRouteState} />;
+      }
+    }, [memoizedRouteState]);
+
+    const handleOpenCreateFlyoutClick = useCallback(() => {
+      setUrlParams({ show: 'create' });
+    }, [setUrlParams]);
+
+    const handlePaginationChange: ArtifactListPagePaginatedContentComponent['onChange'] =
+      useCallback(
+        ({ pageIndex, pageSize }) => {
+          setUrlParams({
+            page: pageIndex + 1,
+            pageSize,
+          });
+
+          // Scroll to the top to ensure that when new set of data is received and list updated,
+          // the user is back at the top of the list
+          window.scrollTo(0, 0);
+        },
+        [setUrlParams]
+      );
+
+    const handleOnSearch = useCallback(
+      (filterValue: string, selectedPolicies: string, doHardRefresh) => {
+        setUrlParams({
+          // `undefined` will drop the param from the url
+          filter: filterValue.trim() === '' ? undefined : filterValue,
```

Comment:
not sure what you mean. filter value and selected polices are represented in the URL with two different params. This is just trying to ensure we clean up the URL params that are not necessary to be there 

_(FYI: my goal is to have a `useUrlParams()` type of hook that can also do that with default values - example: drop them from the URL if the value is set to a default value)_

Reply:
Of course! I meant if we should clean up the URL params when they are not '', as 
```
filterValue.trim() === '' ? undefined : filterValue.trim()
```

Reply:
Ohhhh. I see. 🤔 

thats interesting. I mean, having a string start/end with a space might be a valid filter value. but just having a empty space I don't see as a valid filter and the only reason I'm `.trim()`'ing it is because if we don't, a user could see no results in the list and not really understand why since there would be (visually) nothing in the filter bar